app/domain/people_skills_filter.rb

```diff
@@ -1,21 +1,45 @@
 # frozen_string_literal: true
 
 class PeopleSkillsFilter
-  attr_reader :entries, :rated, :level, :interest
+  attr_reader :entries, :rated, :levels_and_interests_for_skills
 
-  def initialize(entries, rated, level = nil, interest = nil)
+  # rubocop:disable Metrics/ParameterLists
+  def initialize(entries, rated, levels = [], interests = [], skill_ids = [])
     @entries = entries
     @rated = rated
-    @level = level
-    @interest = interest
+    @levels_and_interests_for_skills = []
+    set_levels_and_interests_for_skills(levels, interests, skill_ids)
   end
+  # rubocop:enable Metrics/ParameterLists
 
   def scope
     filter_by_level_and_interest(filter_by_rated)
   end
 
   private
 
+  def set_levels_and_interests_for_skills(levels, interests, skill_ids)
+    return if levels.blank? || skill_ids.blank? || interests.blank?
+
+    levels = levels.to_s.split(',')
+    interests = interests.to_s.split(',')
+    skill_ids = skill_ids.to_s.split(',')
+
+    check_query_param_length(levels, interests, skill_ids)
+
+    # rubocop:disable Layout/LineLength
+    @levels_and_interests_for_skills = skill_ids.zip(levels, interests)
+                                                .map { |search_param| { skill: search_param[0], level: search_param[1], interest: search_param[2] } }
+    # rubocop:enable Layout/LineLength
+  end
+
+  def check_query_param_length(levels, interests, skill_ids)
+    # & is the save navigation operator and checks for nil
+    if levels&.length != skill_ids&.length || interests&.length != skill_ids&.length
```

Comment:
When would this ever be the case?

Reply:
It was implemented by the PSE students already, I already thought about removing it because this case shouldn't exist, should I remove it?

Reply:
Yes do it :grin: